orion/pkg/controllers/class_controller_test.go

```diff
@@ -0,0 +1,555 @@
+	assert.EqualValues(t, later2, classes[1].EndDate)
+
+	assert.EqualValues(t, 2, len(classes))
+}
+
+//
+// Test Get Class
+//
+func TestGetClass_Success(t *testing.T) {
+	now := time.Now().UTC()
+	later := now.Add(time.Hour * 24 * 60)
+	classService.mockGetByClassId = func(classId string) (domains.Class, error) {
+		class := createMockClass(
+			"program1",
+			"2020_spring",
+			"class1",
+			"program1_2020_spring_class1",
+			"churchill",
+			"3 pm - 5 pm",
+			now,
+			later,
+		)
+		return class, nil
+	}
+	services.ClassService = &classService
+
+	// Create new HTTP request to endpoint
+	recorder := sendHttpRequest(t, http.MethodGet, "/api/classes/v1/class/program1_2020_spring_class1", nil)
+
+	// Validate results
+	assert.EqualValues(t, http.StatusOK, recorder.Code)
+	var class domains.Class
+	if err := json.Unmarshal(recorder.Body.Bytes(), &class); err != nil {
+		t.Errorf("unexpected error: %v\n", err)
+	}
+	assert.EqualValues(t, "program1", class.ProgramId)
+	assert.EqualValues(t, "2020_spring", class.SemesterId)
+	assert.EqualValues(t, "class1", class.ClassKey)
+	assert.EqualValues(t, "program1_2020_spring_class1", class.ClassId)
+	assert.EqualValues(t, "churchill", class.LocationId)
+	assert.EqualValues(t, "3 pm - 5 pm", class.Times)
+	assert.EqualValues(t, now, class.StartDate)
+	assert.EqualValues(t, later, class.EndDate)
+}
+
+func TestGetClass_Failure(t *testing.T) {
+	classService.mockGetByClassId = func(classId string) (domains.Class, error) {
+		return domains.Class{}, errors.New("not found")
+	}
+	services.ClassService = &classService
+
+	// Create new HTTP request to endpoint
+	recorder := sendHttpRequest(t, http.MethodGet, "/api/classes/v1/class/program2_2020_summer", nil)
+
+	// Validate results
+	assert.EqualValues(t, http.StatusNotFound, recorder.Code)
+}
+
+//
+// Test Get Classes by other properties
+//
+func TestGetClassesByProgram_Success(t *testing.T) {
+	now := time.Now().UTC()
+	later1 := now.Add(time.Hour * 24 * 30)
+	later2 := now.Add(time.Hour * 24 * 60)
+	later3 := now.Add(time.Hour * 24 * 90)
+	classService.mockGetByProgramId = func(programId string) ([]domains.Class, error) {
+		return []domains.Class{
+			{
+				Id:         1,
+				ProgramId:  "program1",
+				SemesterId: "2020_spring",
+				ClassKey:   "class1",
+				ClassId:    "program1_2020_spring_class1",
+				LocationId: "churchill",
+				Times:      "3 pm - 5 pm",
+				StartDate:  now,
+				EndDate:    later1,
+			},
+			{
+				Id:         2,
+				ProgramId:  "program2",
+				SemesterId: "2020_summer",
+				ClassKey:   "",
+				ClassId:    "program2_2020_summer",
+				LocationId: "churchill",
+				Times:      "5 pm - 7 pm",
+				StartDate:  now,
+				EndDate:    later2,
+			},
+			{
+				Id:         3,
+				ProgramId:  "program1",
+				SemesterId: "2020_fall",
+				ClassKey:   "class2",
+				ClassId:    "program1_2020_fall_class2",
+				LocationId: "churchill",
+				Times:      "5 pm - 7 pm",
+				StartDate:  now,
+				EndDate:    later3,
+			},
```

Comment:
consider making this a global variable or from a `createFakeClasses()` helper function which you can then use for these 3-4 similar tests (maybe even for all the tests if you'd like).
Make the mock function return the result from `createFakeClasses()` or global variable and do the rest of the testing in each unit tests!